SeaPkg/Library/BasePeCoffValidationLib/BasePeCoffValidationLib.c

```diff
@@ -460,7 +461,9 @@ PeCoffImageValidationPointer (
     goto Done;
   }
 
-  InMseg = ((UINTN)((UINT8 *)TargetImage + Hdr->Offset) >= MsegBase) && ((UINTN)((UINT8 *)TargetImage + Hdr->Offset) < MsegBase + MsegSize);
+  AddrInTarget = 0;
+  CopyMem (&AddrInTarget, (UINT8 *)TargetImage + Hdr->Offset, Hdr->Size);

```

Comment:
Is this meant to copy `Hdr->Size` bytes? `AddrInTarget` is a `EFI_PHYSICAL_ADDRESS`. This code would also need to check the value of `Hdr->Size`...

I may be misunderstanding but it seems like you want to set `AddrInTarget` to the address of that offset?

```suggestion
  AddrInTarget = ((EFI_PHYSICAL_ADDRESS)(UINTN)((UINT8 *)TargetImage + Hdr->Offset));
```

Reply:
Clarified offline that this is the resolved address in MSEG at this location and the size of the content is verified earlier against the Hdr->Size.